### PR TITLE
Add PlayerPickupItemEvent

### PR DESCRIPTION
**Pull request by MCBEPluginModer**
Add PlayerPickupItemEvent

### CODE REVIEW

**Comment by KawaiiNahida** [comment]
Player::take should not trigger PlayerDropItemEvent  event

**Review by KawaiiNahida — CHANGES_REQUESTED** [review]
Err in Line 1960,1968

File: LiteLoader/Kernel/EventAPI.cpp [hunk 1]
```diff
@@ -1952,4 +1953,18 @@ THook(void, "?dropSlot@Inventory@@QEAAXH_N00@Z", Container* a1, int a2, char a3,
         IF_LISTENED_END(PlayerDropItemEvent)
     }
     return original(a1, a2, a3,a4,a5);
-}
\ No newline at end of file
+}
+
+THook(char, "?take@Player@@QEAA_NAEAVActor@@HH@Z", Player* a1, Actor& a2, int a3, int a4)
```
Comment (Jaciezyt) [inline LiteLoader/Kernel/EventAPI.cpp:1958]:
```suggestion
TInstanceHook(char, "?take@Player@@QEAA_NAEAVActor@@HH@Z", Player, Actor& a2, int a3, int a4)
```
Please use `TInstanceHook` like above

**Comment by yqs112358** [comment]
Emm anyway thanks for your comtribution
However, we have actually had this kind of event below, which is named "PlayerTakeItem"
I admit that its name is not proper now, and we will rename it later

**Pull request closed without merging**